presto-main/src/main/java/io/prestosql/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -514,6 +671,41 @@ else if (writerTarget instanceof CreateReference || writerTarget instanceof Inse
             return processChildren(node, context);
         }
 
+        private EstimatedStatsAndCost getEstimatedStatsAndCost(TableScanNode node, TupleDomain<ColumnHandle> predicate, TableMetadata tableMetadata)
+        {
+            QualifiedObjectName tableName = new QualifiedObjectName(
+                    tableMetadata.getCatalogName().getCatalogName(),
+                    tableMetadata.getTable().getSchemaName(),
+                    tableMetadata.getTable().getTableName());
+            Optional<TableHandle> optTableHandle = metadata.getTableHandle(session, tableName);
+            double bytesScanned = 0;
+            if (optTableHandle.isPresent()) {
+                TableHandle table = optTableHandle.get();
+                Constraint constraint = new Constraint(predicate);
+                TableStatistics tableStatistics = metadata.getTableStatistics(session, table, constraint);
```

Comment:
Getting table statistics could be potentially expensive. It would be nice to leverage only information from `StatsAndCosts`. Could first expose estimates without `bytesScanned` statistic.

Also why `stats.getOutputSizeInBytes()` is not enough? What is the difference between `stats.getOutputSizeInBytes()` and `bytesScanned`? IMO they are the same thing.